nautobot_golden_config/forms.py

```diff
@@ -549,15 +548,14 @@ class Meta:
 class ConfigPlanUpdateForm(NautobotModelForm):
     """Form for ConfigPlan instances."""
 
-    change_control_id = forms.CharField(required=False, label="Change Control ID")
-    change_control_url = forms.URLField(required=False, label="Change Control URL")
+    change_control_id = forms.CharField(label="Change Control ID")
+    change_control_url = forms.URLField(label="Change Control URL")
     status = utilities_forms.DynamicModelChoiceField(
         queryset=Status.objects.all(),
         query_params={"content_types": models.ConfigPlan._meta.label_lower},
-        required=False,
     )
     tag = utilities_forms.DynamicModelMultipleChoiceField(
-        queryset=Tag.objects.all(), query_params={"content_types": "dcim.device"}, required=False
+        queryset=Tag.objects.all(), query_params={"content_types": "dcim.device"}
```

Comment:
I'm not sure why these were removed. This makes it so you can't edit a Config Plan without these items now.

![Screenshot 2023-09-11 at 9 54 49 AM](https://github.com/nautobot/nautobot-plugin-golden-config/assets/10467633/cca6e9a3-c11b-42d6-a898-954cbc56c91a)

Reply:
ok, need to figure this out... I mentioned in the other note my reasoning, but now need to ask the wise council :)

Reply:
ok, had to flush it out with the help from @glennmatthews, basically:

* All explicit form class attributes override the magic that happens when you define everything in fields, so required=false is required in those circumstances
* Nested fields need an explicit form class attribute
* Model fields/attributes generally should not need explicit form class attributes (but there will be reasons of course)
    * Do note that you do not need label if you have verbose_name on the model, so that is not a reason to have an explicit attribute on the model, it is better centralize that value on the model attribute
* Ordering should generally be set on the model, or more specifically, the order in which migrations were added and ordered
    * You can control with `field_order` if need to be explicit
* fields should generally be `__all__`, which is the combo of attributes on the form class plus attributes on the model class

Reply:
so this is fully resolved now?

Reply:
Correct.